services/skus/service.go

```diff
@@ -1958,27 +1947,22 @@ func (s *Service) redeemBlindedCred(ctx context.Context, w http.ResponseWriter,
 	// FIXME: we shouldn't be using the issuer as the payload, it ideally would be a unique request identifier
 	// to allow for more flexible idempotent behavior.
 	if err := redeemFn(ctx, cred.Issuer, cred.TokenPreimage, cred.Signature, cred.Issuer); err != nil {
-		msg := err.Error()
-
-		// Time limited v2: Expose a credential id so the caller can decide whether to allow multiple redemptions.
-		if kind == timeLimitedV2 && msg == cbr.ErrDupRedeem.Error() {
-			data := &blindedCredVrfResult{ID: cred.TokenPreimage, Duplicate: true}
-
-			return handlers.RenderContent(ctx, data, w, http.StatusOK)
+		if !shouldRetryRedeemFn(kind, cred.Issuer, err) {
+			return handleRedeemFnError(ctx, w, kind, cred, err)
 		}
 
-		// Duplicate redemptions are not verified.
-		if msg == cbr.ErrDupRedeem.Error() || msg == cbr.ErrBadRequest.Error() {
-			return handlers.WrapError(err, "invalid credentials", http.StatusForbidden)
+		// Fix for https://github.com/brave-intl/challenge-bypass-server/pull/371.
+		const leoa = "brave.com?sku=brave-leo-premium-year"
+		if err := redeemFn(ctx, leoa, cred.TokenPreimage, cred.Signature, leoa); err != nil {
```

Comment:
```suggestion
		if err := redeemFn(ctx, leoa, cred.TokenPreimage, cred.Signature, cred.Issuer); err != nil {
```

it struck me as odd that this function needs to have the issuer passed as two separate parameters. on looking at it, it seems that the latter is the `payload` parameter - which is a value the user is binding the credential use to via a computed MAC. 

we really should not be using the issuer there but putting that aside, we need to ensure we use the value that the client computed the MAC over. in this case that would be the same issuer the client believes they are using

Reply:
Thanks, that's what I thought as well as discussed in DM. Updated.